core/src/client/lwm2m_static.c

```diff
@@ -401,6 +488,11 @@ AwaError AwaStaticClient_RegisterResourceWithHandler(AwaStaticClient * client, c
             ResourceDefinition * resourceDefinition = Definition_NewResourceTypeWithHandler(objFormat, resourceName, resourceID, resourceType, minimumInstances, maximumInstances, operations, (LWM2MHandler)handler);
             if (resourceDefinition != NULL)
             {
+                // TODO: perhaps a better option would be to bind some opaque data
```

Comment:
The definition registry maintains a pointer to the data, and two integers to support each resource. I'm not sure this is the best approach. I was thinking perhaps I could allocate some memory in lwm2m_static to store this data when required and store a pointer to this in the definition registry instead, what do you guys think?